surveys/2023-annual-survey/questions.md

```diff
@@ -0,0 +1,769 @@
+> change depending on the size of the development effort at a company.
+>
+> Previously this question used "employees" instead of "developers". It is more appropriate for us to ask about the amount
+> of developers at a company vs. the amount of people employed in total.
+
+### Is your company planning on hiring Rust developers in the next year?
+
+Type: select one (optional)
+
+- Yes
+- No
+- I don't know
+
+> This question assess hiring sentiment. Although there is intrinsic uncertainty, it is easy to answer and forward looking.
+> It will also be interesting to see what the demand for Rust skills from companies is over time.
+
+## Your opinions about Rust
+
+### Which of the following statements about Rust do you feel are true?
+
+Type: select all that apply (optional)
+
+Statements:
+
+- Rust provides a real benefit over other programming languages
+- Rust is significantly more complicated to program in than other programming languages
+- Rust requires significantly more effort to learn than other programming languages
+- Rust code tends to contain significantly fewer bugs than equivalent code written in another programming language would have
+- Rust is risky to use in production
+- Rust makes me more productive
+- Rust is fun to use
+
+> **justification**
+>
+> There are several "truisms" about Rust that we'd like to get perspective on how true these are for users of Rust.
+>
+> Note that answers here can be subject to survivorship bias and so extra care should be taken with interpreting results.
+
+
+### What are your biggest worries for the future of Rust?
+
+Type: select all that apply (optional)
+
+- Not enough usage in industry
+- Too much interest from big companies
+- Not enough open source contributions to the ecosystem
+- Doesn't add a specific feature I want
+- Rust doesn't evolve quickly enough
+- Instability of the language
+- Superseded by an alternative
+- Becomes too complex
+- Tools and documentation are not accessible enough (e.g., due to language or incompatibility with screen readers)
+- Project governance does not scale to match the size/requirements of the community
+- Developers/maintainers of the language are not properly supported
+- I'm not worried
+- Other (open response)
+
+> **justification**
+>
+> Would be useful for leadership to understand the community's fears.
+
+### In your opinion, how would you prioritise work on the following aspects of Rust?
+
+Type: ordered list (optional)
+
+Aspects:
+
+- Runtime performance
+- Compile times
+- Binary size
+- Memory usage (i.e., how much RAM rustc uses when compiling)
+- Disk space usage (e.g., the size of target folder)
+- Bugs in the compiler (i.e., ICEs a.k.a. internal compiler errors, miscompilations, etc.)
+- Compiler error messages
+- IDE experience
+- Debugging experience
+- Available tools and support
+- Async programming
+- GUI development
```

Comment:
This is a weird outlier here. Unlike the other options, this isn't really the Rust project's responsibility in any way.

Reply:
agreed. could be moved elsewhere (if not already included as an option of another question)

Reply:
I agree that the async/GUI answers are outliers, I have removed them.